src/OneExplorer/OneStorage.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+import {obtainWorkspaceRoot} from '../Utils/Helpers';
+import {Logger} from '../Utils/Logger';
+
+import {ConfigObj} from './ConfigObject';
+
+interface StringMap {
+  [key: string]: string[];
+}
+
+interface ConfigObjMap {
+  [key: string]: ConfigObj|null;
+}
+
+/**
+ * A singleton storage class
+ *
+ * PURPOSE
+ *
+ * To build each 'Node' of OneTreeDataProvider,
+ * it is neccessary to access the file system, read the files and build objects(ConfigObj, ...).
+ * By keeping some file system information as data structure (list, map),
+ * some duplicated works can be reduced.
+ *
+ * LIFE CYCLE
+ *
+ * The singleton is created when the first get() is called.
+ * The object remains until OneStorage.reset() is called.
+ * OneStorage.reset() is called by OneTreeDataProvider.refresh(), which is called on every file
+ * system change within the repository.
+ */
+export class OneStorage {
+  /**
+   * A list of all cfg paths in the 'root' directory
+   */
+  private _cfgList: string[];
+  /**
+   * A map of ConfigObj (key: cfg path)
+   */
+  private _cfgToCfgObjMap: ConfigObjMap;
+  /**
+   * A map of BaseModel path to Cfg path
+   */
+  private _baseModelToCfgsMap: StringMap;
+
+  /**
+   * Get the list of .cfg files within the workspace
+   * @param root  the file or directory,
+   *              which MUST exist in the file system
+   */
+  private _getCfgList(root: string = obtainWorkspaceRoot()): string[] {
+    /**
+     * Returns every file inside directory
+     * @todo Check soft link
+     * @param root
+     * @returns
+     */
+    const readdirSyncRecursive = (root: string): string[] => {
+      if (fs.statSync(root).isFile()) {
+        return [root];
+      }
+
+      let children: string[] = [];
+      if (fs.statSync(root).isDirectory()) {
+        fs.readdirSync(root).forEach(val => {
+          children = children.concat(readdirSyncRecursive(path.join(root, val)));
+        });
+      }
+      return children;
+    };
+
+    try {
+      fs.statSync(root);
+    } catch {
+      Logger.error('OneExplorer', 'getCfgList', 'called on not existing directory or file.');
+      return [];
+    }
+
+    return readdirSyncRecursive(root).filter(val => val.endsWith('.cfg'));
```

Comment:
(Optional) After I saw `fs.statSync` is in try-catch, I thought following suggestion because `readdirSyncRecursive` also includes `fs.statSync` :)
But not that critical I think :D

```suggestion
    try {
      fs.statSync(root);
      return readdirSyncRecursive(root).filter(val => val.endsWith('.cfg'));
    } catch {
      Logger.error('OneExplorer', 'getCfgList', 'called on not existing directory or file.');
      return [];
    }
```

Reply:
I agree :-D Then it seems the line `fs.statSync(root);` can be removed, too. Thanks :-D